plugins/modules/podman_login.py

```diff
@@ -0,0 +1,160 @@
+    If the registry is not specified, the first registry under
+    `[registries.search]` from `registries.conf `will be used. The path of
+    the authentication file can be overridden by the user by setting the
+    `authfile` flag. The default path used is
+    `${XDG_RUNTIME_DIR}/containers/auth.json`.
+requirements:
+  - "Podman installed on host"
+options:
+  authfile:
+    description:
+      - Path of the authentication file. Default is
+        ``${XDG_RUNTIME_DIR}/containers/auth.json``
+        You can also override the default path of the authentication
+        file by setting the ``REGISTRY_AUTH_FILE`` environment
+        variable. ``export REGISTRY_AUTH_FILE=path``
+    type: path
+  certdir:
+    description:
+      - Use certificates at path (*.crt, *.cert, *.key) to connect
+        to the registry.  Default certificates directory
+        is /etc/containers/certs.d.
+    type: path
+  password:
+    description:
+      - Password for the registry server.
+    required: True
+    type: str
+  registry:
+    description:
+      - Registry server. If the registry is not specified,
+        the first registry under `[registries.search]` from
+        `registries.conf` will be used.
+    type: str
+  tlsverify:
+    description:
+      - Require HTTPS and verify certificates when
+        contacting registries.  If explicitly set to true,
+        then TLS verification will be used. If set to false,
+        then TLS verification will not be used.  If not specified,
+        TLS verification will be used unless the target registry
+        is listed as an insecure registry in registries.conf.
+    type: bool
+  username:
+    description:
+      - Username for the registry server.
+    required: True
+    type: str
+  executable:
+    description:
+      - Path to C(podman) executable if it is not in the C($PATH) on the
+        machine running C(podman)
+    default: 'podman'
+    type: str
+'''
+
+EXAMPLES = r"""
+- name: Login to default registry and create ${XDG_RUNTIME_DIR}/containers/auth.json
+  containers.podman.podman_login:
+    username: user
+    password: 'p4ssw0rd'
+
+- name: Login to default registry and create ${XDG_RUNTIME_DIR}/containers/auth.json
+  containers.podman.podman_login:
+    username: user
+    password: 'p4ssw0rd'
+    registry: quay.io
+
+"""
+# noqa: F402
+
+import json  # noqa: F402
+from ansible.module_utils.basic import AnsibleModule
+
+
+def login(module, executable, registry, authfile,
+          certdir, tlsverify, username, password):
+
+    command = [executable, 'login']
+    changed = False
+
+    if username:
+        command.extend(['--username', username])
+    if password:
+        command.extend(['--password', password])
+    if authfile:
+        command.extend(['--authfile', authfile])
+    if registry:
+        command.append(registry)
+    if certdir:
+        command.extend(['--cert-dir', certdir])
+    if tlsverify:
```

Comment:
With booleans we should be careful and check if they are not `None`, otherwise `tlsverify=False` will never be configured.
```suggestion
    if tlsverify is not None:
```

Reply:
Good catch, thank you